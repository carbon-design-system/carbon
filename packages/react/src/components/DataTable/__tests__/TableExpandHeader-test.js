/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import { Table, TableHead, TableRow, TableExpandHeader } from '../';

describe('DataTable.TableExpandHeader', () => {
  it('should render', () => {
    const wrapper = mount(
      <Table>
        <TableHead>
          <TableRow>
            <TableExpandHeader
              ariaLabel="test-label"
              isExpanded={false}
              onExpand={jest.fn()}
              className="custom-class"
            />
          </TableRow>
        </TableHead>
      </Table>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
