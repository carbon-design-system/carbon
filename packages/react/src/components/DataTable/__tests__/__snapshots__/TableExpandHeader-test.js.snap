// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableExpandHeader should render 1`] = `
<Table
  isSortable={false}
  overflowMenuOnHover={true}
>
  <div
    className="bx--data-table-content"
  >
    <table
      className="bx--data-table bx--data-table--no-border"
    >
      <TableHead>
        <thead>
          <TableRow>
            <tr>
              <TableExpandHeader
                ariaLabel="test-label"
                className="custom-class"
                isExpanded={false}
                onExpand={[MockFunction]}
              >
                <th
                  className="bx--table-expand custom-class"
                  scope="col"
                />
              </TableExpandHeader>
            </tr>
          </TableRow>
        </thead>
      </TableHead>
    </table>
  </div>
</Table>
`;
